src/DotNetWorker.Core/Diagnostics/FunctionActivitySource.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Diagnostics
+{
+    internal class FunctionActivitySourceFactory
+    {
+        private static readonly ActivitySource _activitySource = new(TraceConstants.FunctionsActivitySource);
+        private readonly string _schemaVersionUrl;
+        private readonly Lazy<IReadOnlyDictionary<string, string>> _attributeMap;
+
+        public FunctionActivitySourceFactory(IOptions<WorkerOptions> options)
+        {
+            _attributeMap = new Lazy<IReadOnlyDictionary<string, string>>(() => GetMapping(options.Value.OpenTelemetrySchemaVersion));
+            _schemaVersionUrl = TraceConstants.OpenTelemetrySchemaMap[options.Value.OpenTelemetrySchemaVersion];
+        }
+
+        public Activity? StartInvoke(FunctionContext context)
+        {
+            var activity = _activitySource.StartActivity(TraceConstants.FunctionsInvokeActivityName, ActivityKind.Internal, context.TraceContext.TraceParent,
+                tags: GetTags(context));
+
+            return activity;
```

Comment:
would be great to add `activity.Tracestate = context.TraceContext.Tracestate`

Reply:
Done.